gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -401,18 +405,24 @@ Builder setProcessorCount(int processorCount) {
     }
 
     /**
-     * Sets the ExecutorProvider for this TransportChannelProvider.
+     * Sets the Executor for this TransportChannelProvider.
      *
      * <p>This is optional; if it is not provided, needsExecutor() will return true, meaning that an
      * Executor must be provided when getChannel is called on the constructed
      * TransportChannelProvider instance. Note: GrpcTransportProvider will automatically provide its
      * own Executor in this circumstance when it calls getChannel.
      */
-    public Builder setExecutorProvider(ExecutorProvider executorProvider) {
-      this.executorProvider = executorProvider;
+    public Builder setExecutor(Executor executor) {
+      this.executor = executor;
       return this;
     }
 
+    /** @deprecated. Please use {@link #setExecutor(Executor)}. */
+    @Deprecated
+    public Builder setExecutor(ScheduledExecutorService executor) {
```

Comment:
This should be called `setExecutorProvider` and accept ExecutorProvider as an argument (i.e. put back the old method)

Reply:
Done.